janitor/functions/pivot.py

```diff
@@ -313,13 +359,7 @@ def _data_checks_pivot_longer(
 
         if isinstance(names_pattern, (list, tuple)):
```

Comment:
Shall we convert this into an `elif`? The `if` before this is mutually exclusive with this `if` check.

```suggestion
        elif isinstance(names_pattern, (list, tuple)):
```